Movecraft/src/main/java/net/countercraft/movecraft/commands/MovecraftCommandManager.java

```diff
@@ -0,0 +1,61 @@
+package net.countercraft.movecraft.commands;
+
+import co.aikar.commands.CommandIssuer;
+import co.aikar.commands.PaperCommandManager;
+import org.bukkit.command.CommandSender;
+import org.bukkit.entity.Player;
+import org.bukkit.plugin.Plugin;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.regex.Pattern;
+
+public class MovecraftCommandManager extends PaperCommandManager {
+    public MovecraftCommandManager(Plugin plugin) {
+        super(plugin);
+    }
+
+    private static final Pattern COMMA = Pattern.compile(",");
+    private static final Pattern PIPE = Pattern.compile("\\|");
+
+    @Override
+    public boolean hasPermission(CommandIssuer issuer, String permission) {
+        if (permission == null || permission.isEmpty()) {
+            return true;
+        }
+
+        //handle AND like normal using comma ","
+        String[] perms = COMMA.split(permission);
+        if (perms.length > 1) {
+            return super.hasPermission(issuer, new HashSet<>(Arrays.asList(perms)));
+        }
+
+        //handle OR using pipe "|"
+        CommandSender sender = issuer.getIssuer();
+        for (String perm : PIPE.split(permission)) {
+            perm = perm.trim();
+            if (!perm.isEmpty() && senderHasPermission(sender, perm)) {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    public static boolean senderHasPermission(@NotNull CommandSender sender, @NotNull String perm) {
+        if (!(sender instanceof Player p)) {
+            return sender.isOp();
+        }
+
+        if (p.isOp())
+            return true;
+
+        if (p.hasPermission(perm) || p.hasPermission("movecraft.all")) {
+            return true;
+        }
+
+        return false;
+    }
```

Comment:
I don't think we want to complicate things by adding special cases for op or "movecraft.all", since this code is only used by commands.

Reply:
I can get rid of the `movecraft.all` that one was my decision, not sure what you mean with special case for op if anything this makes things easier, if player is op it doesn't have to do any permission checks, and if it is not a player then it just check if it is op like console (for the few commands that can be ran from console that is)

Reply:
> I can get rid of the `movecraft.all` that one was my decision, not sure what you mean with special case for op if anything this makes things easier, if player is op it doesn't have to do any permission checks, and if it is not a player then it just check if it is op like console (for the few commands that can be ran from console that is)

Dont check for op here! That is handled by bukkits perm system on its own by specifying the permission in the permission.yml (or was it somewhere in plugin.yml?) file. If you set the perm to op by default, all operators will have it. Also you should not override and effectively take away from permission systems like that!

Reply:
Also if it is console, iirc there is some other way to check that on the sender instance

Reply:
Also movecraft.all is unnecessary. Permission systems normally allow things like movecraft.* which covers all movecraft.<anything> permissions. No need to implement that on your own in addition, that just bloats up the codebase and adds unnecessary clutter code

Reply:
> > I can get rid of the `movecraft.all` that one was my decision, not sure what you mean with special case for op if anything this makes things easier, if player is op it doesn't have to do any permission checks, and if it is not a player then it just check if it is op like console (for the few commands that can be ran from console that is)
> 
> Dont check for op here! That is handled by bukkits perm system on its own by specifying the permission in the permission.yml (or was it somewhere in plugin.yml?) file. If you set the perm to op by default, all operators will have it. Also you should not override and effectively take away from permission systems like that!

the whole point of this is getting rid of the plugin.yml spigot format to use the new paper-plugin.yml that is why i can't just throw it there

Reply:
Unless paper-plugin.yml handles permissions too? let me check

Reply:
I am pretty sure they have something similar, would be weird if they reduce features like that as that could break multiple servers in a way

Reply:
One of the main dev of paper said it is handled the same way probably, so i can get rid of the player op check, not sure about the console one as idk any alternative

Reply:
Man I remembered CommandSender didn't have an `hasPermission(String)` method, it looks like i was wrong, there is no need for all this stuff after all